lib/tools/apk-utils.js

```diff
@@ -0,0 +1,164 @@
+    let thirdparty = apiLevel >= 15 ? "-3" : "";
+    let stdout = await this.shell(['pm', 'list', 'packages', thirdparty, pkg]);
+    let apkInstalledRgx = new RegExp(`^package:${pkg.replace(/(\.)/g, "\\$1")}$`,
+                                     'm');
+    installed = apkInstalledRgx.test(stdout);
+    log.debug(`App is ${!installed ? " not" : ""} installed`);
+    return installed;
+  } catch (e) {
+    log.errorAndThrow(`Error finding if app is installed. Original error: ${e.message}`);
+  }
+};
+
+apkUtilsMethods.startApp = async function (startAppOptions = {}) {
+  try {
+    if (!startAppOptions.activity || !startAppOptions.pkg) {
+      log.errorAndThrow("activity and pkg is required for launching application");
+    }
+    startAppOptions = _.clone(startAppOptions);
+    // initializing defaults
+    _.defaults(startAppOptions, {
+        waitPkg: startAppOptions.pkg,
+        waitActivity: false,
+        retry: true,
+        stopApp: true
+    });
+    // preventing null waitpkg
+    startAppOptions.waitPkg = startAppOptions.waitPkg || startAppOptions.pkg;
+    let apiLevel = await this.getApiLevel();
+    let cmd = buildStartCmd(startAppOptions, apiLevel);
+    let stdout = await this.shell(cmd);
+    if (stdout.indexOf("Error: Activity class") !== -1 &&
+        stdout.indexOf("does not exist") !== -1) {
+      if (startAppOptions.retry && startAppOptions.activity[0] !== ".") {
+        log.debug("We tried to start an activity that doesn't exist, " +
+                  "retrying with . prepended to activity");
+        startAppOptions.activity = "." + startAppOptions.activity;
+        startAppOptions.retry = false;
+        return this.startApp(startAppOptions);
+      } else {
+        log.errorAndThrow("Activity used to start app doesn't exist or cannot be " +
+                          "launched! Make sure it exists and is a launchable activity");
+      }
+    } else if (stdout.indexOf("java.lang.SecurityException") !== -1) {
+      // if the app is disabled on a real device it will throw a security exception
+      log.errorAndThrow("Permission to start activity denied.");
+    }
+    if (startAppOptions.waitActivity) {
+      if (startAppOptions.hasOwnProperty("waitDuration")) {
+        await this.waitForActivity(startAppOptions.waitPkg, startAppOptions.waitActivity,
+                                   startAppOptions.waitDuration);
+      } else {
+        await this.waitForActivity(startAppOptions.waitPkg, startAppOptions.waitActivity);
+      }
+    }
+  } catch (e) {
+    log.errorAndThrow(`Error occured while starting App. Original error: ${e.message}`);
+  }
+};
+
+
+apkUtilsMethods.getFocusedPackageAndActivity = async function () {
+  log.debug("Getting focused package and activity");
+  let cmd = ['dumpsys', 'window', 'windows'],
+      nullRe = new RegExp(/mFocusedApp=null/),
+      searchRe = new RegExp(
+        /mFocusedApp.+Record\{.*\s([^\s\/\}]+)\/([^\s\/\}]+)(\s[^\s\/\}]+)*\}/);
```

Comment:
yes I tried that but i would need to escape all `/` which was looking but odd with too many '//'